Returning company information

s-e-v opened the pull request:
The "userIdentifier" function returns the company information based on the user's token post login.

The tests check for various scenarios (the function does not check if the token is not available as the user would be unable to be on the page in the first place).

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
     });
   }
 
+  userIdentifier() {
```
I think this is a bad name. How about `getUserCompany()`?

s-e-v replied:
Good idea, implemented.

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
 
+  userIdentifier() {
+    try {
+      const token = Cookies.get(this._envConfig.SSO_COOKIE_NAME);
```
This should call `this._getTokenCookie()`

s-e-v replied:
Now it works again, fixed.

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,17 @@ class ANDGatewayClient {
+      const company = payload["http://sso.and.digital/company"];
+      return company;
+    } catch (error) {
+      return true;
```
Is `true` a good default value for company?

s-e-v replied:
Fixed, please check the new code.

oskanberg commented on test/unit.js:
```diff
@@ -127,3 +127,32 @@ describe("getExpiryDate", () => {
+  it("should return undefined if token is empty", () => {
+    const token = "";
+    client._setTokenCookie(token);
+    expect(client.userIdentifier()).toBe(undefined);
```
Really? As it is, doesn't it return `true`?

s-e-v replied:
Fixed, please check the new code.

s-e-v commented:
Please check the new code/tests:
1) Renamed the function
2) Removed the promise (unnecessary)
3) 2 IFs to check at different stages if the token is invalid, and if the company key exists in the token, otherwise return the company info
4) Tests updated to match the new changes

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
+  getUserCompany() {
+    const token = this._getTokenCookie();
+    if (token == undefined) {
+      return "No token found/Token is not valid";
```
how about `throw new Error('Authentication token not found')`

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
+      const payload = decodeJWT(token);
+      const company = payload["http://sso.and.digital/company"];
+      if (company == undefined) {
+        return "The value does not exist in the token";
```
I think that returning undefined or '' works here

oskanberg commented on src/index.js:
```diff
@@ -61,6 +61,21 @@ class ANDGatewayClient {
     });
   }
 
+  getUserCompany() {
```
Please add JSDoc comment (like the other methods) and rebuild - this will automatically add it to the readme.
